Makefile

```diff
@@ -1,59 +1,61 @@
+# - use the VERSION as arg of the bundle target (e.g make bundle VERSION=0.0.2)
+# - use environment variables to overwrite this value (e.g export VERSION=0.0.2)
+VERSION ?= master
+
+# CHANNELS define the bundle channels used in the bundle.
+# Add a new line here if you would like to change its default config. (E.g CHANNELS = "preview,fast,stable")
+# To re-generate a bundle for other specific channels without changing the standard setup, you can:
+# - use the CHANNELS as arg of the bundle target (e.g make bundle CHANNELS=preview,fast,stable)
+# - use environment variables to overwrite this value (e.g export CHANNELS="preview,fast,stable")
+ifneq ($(origin CHANNELS), undefined)
+BUNDLE_CHANNELS := --channels=$(CHANNELS)
+endif
+
+# DEFAULT_CHANNEL defines the default channel used in the bundle.
+# Add a new line here if you would like to change its default config. (E.g DEFAULT_CHANNEL = "stable")
+# To re-generate a bundle for any other default channel without changing the default setup, you can:
+# - use the DEFAULT_CHANNEL as arg of the bundle target (e.g make bundle DEFAULT_CHANNEL=stable)
+# - use environment variables to overwrite this value (e.g export DEFAULT_CHANNEL="stable")
+ifneq ($(origin DEFAULT_CHANNEL), undefined)
+BUNDLE_DEFAULT_CHANNEL := --default-channel=$(DEFAULT_CHANNEL)
+endif
+BUNDLE_METADATA_OPTS ?= $(BUNDLE_CHANNELS) $(BUNDLE_DEFAULT_CHANNEL)
+
+# IMAGE_TAG_BASE defines the docker.io namespace and part of the image name for remote images.
+# This variable is used to construct full image tags for bundle and catalog images.
 #
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-# Package related
-BINARY_NAME=network-operator
-PACKAGE=network-operator
-ORG_PATH=github.com/Mellanox
-REPO_PATH=$(ORG_PATH)/$(PACKAGE)
-CHART_PATH=$(CURDIR)/deployment/$(PACKAGE)
+# For example, running 'make bundle-build bundle-push catalog-build catalog-push' will build and push both
+# mellanox.com/network-operator-bundle:$VERSION and mellanox.com/network-operator-catalog:$VERSION.
+IMAGE_TAG_BASE ?= mellanox.com/network-operator
+
+# BUNDLE_IMG defines the image:tag used for the bundle.
+# You can use it as an arg. (E.g make bundle-build BUNDLE_IMG=<some-registry>/<project-name-bundle>:<tag>)
+BUNDLE_IMG ?= $(IMAGE_TAG_BASE)-bundle:v$(VERSION)
+
+# Image URL to use all building/pushing image targets
+IMG ?= docker.io/mellanox/network-operator:latest
+# Produce CRDs that work back to Kubernetes 1.11 (no version conversion)
+CRD_OPTIONS ?= "crd:trivialVersions=true,preserveUnknownFields=false"
+
+# Get the currently used golang install path (in GOPATH/bin, unless GOBIN is set)
+ifeq (,$(shell go env GOBIN))
+GOBIN=$(shell go env GOPATH)/bin
```

Comment:
L#51 defines GOPATH if not specified. however here we take the default value.

Reply:
I'm not sure what you're trying to say?
`# Get the currently used golang install path (in GOPATH/bin, unless GOBIN is set)`

Reply:
So, ill address both comments here.

to my understanding, previously GOPATH and GOBIN, unless specifically passed to Makefile would take the value `$(CURDIR)/.gopath`, `$(CURDIR)/bin` respectively these would then be exported to reflect the new values as env variables to be used by go.

this was done to avoid modifying the content of your original `GOPATH` and `GOBIN` paths (i think that was the original intention) when you run make from your dev machine / laptop.


now,
GOPATH is set within makefile (L51) but never exported so `go env GOPATH` and `$GOPATH` might have different values.
GOBIN would either have the `$GOBIN` env value (if set) or `$GOPATH/bin` where `GOPATH` is the current env varable value (it will never take the value for `$(CURDIR)/bin`).


its a change in behaviour. also i see CI faling in finding some targets, may be related ?

Reply:
I don't see any reason why one would want to "not modify the content of your original `GOPATH` and `GOBIN`. That's the expected and current behavior for any go build. The environment variables can be locally overridden at any time with other paths. 

Master doesn't pass CI (lint) either and yet here we are.